h2/src/main/org/h2/expression/function/Function.java

```diff
@@ -899,7 +899,13 @@ private Value getSimpleValue(Session session, Value v0, Expression[] args,
                     database.getMode().treatEmptyStringsAsNull);
             break;
         case IDENTITY:
-            result = session.getLastIdentity();
+            if (args.length == 0) {
+                result = session.getLastIdentity();
+            } else {
+                final Value value = getNullOrValue(session, args, values, 0);
+                session.setLastIdentity(value);
```

Comment:
Use parameter `v0` here.

Also I think that it's reasonable to convert the argument to `Value.LONG` (`v0.convertTo(Value.LONG)`) to ensure that preserved last identity will have a valid data type.

BTW, what exactly MySQL does if `NULL` is passed as an argument?

Reply:
BTW, what exactly MySQL does if NULL is passed as an argument?


MariaDB [test1]> SELECT LAST_INSERT_ID(NULL);
+----------------------+
| LAST_INSERT_ID(NULL) |
+----------------------+
|                 NULL |
+----------------------+
1 row in set (0.00 sec)

MariaDB [test1]> SELECT LAST_INSERT_ID();
+------------------+
| LAST_INSERT_ID() |
+------------------+
|                0 |
+------------------+
1 row in set (0.00 sec)
`
version    10.0.38-MariaDB